enable/trait_defs/ui/kiva_font_editor.py

```diff
@@ -0,0 +1,128 @@
+# (C) Copyright 2005-2022 Enthought, Inc., Austin, TX
+# All rights reserved.
+#
+# This software is provided without warranty under the terms of the BSD
+# license included in LICENSE.txt and may be redistributed only under
+# the conditions described in the aforementioned license. The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+# Thanks for using Enthought open source!
+
+from pyface.font import Font as PyfaceFont
+from pyface.font_dialog import get_font
+from traits.api import Bool, Callable, Instance, Str, observe
+from traits.trait_base import SequenceTypes
+from traitsui.api import EditorFactory, Editor as BaseEditor, toolkit_object
+
+from kiva.fonttools.font import Font
+import kiva.constants as kc
+from enable.component import Component
+from enable.label import Label
+from enable.tools.button_tool import ButtonTool
+from enable.window import Window
+from .editor_with_component import EditorWithLabelComponent
+
+
+def face_name(font):
+    """ Returns a Font's typeface name.
+    """
+    face_name = font.face_name
+    if isinstance(face_name, SequenceTypes):
+        face_name = face_name[0]
+
+    return face_name
+
+
+def str_font(font):
+    """ Returns the text representation of the specified font trait value
+    """
+
+    weight = " Bold" if font.is_bold() else ""
+    style = " Italic" if font.style in kc.italic_styles else ""
+    underline = " Underline" if font.underline else ""
+
+    return f"{font.size} point {face_name(font)}{weight}{style}{underline}".strip()
+
+
+class ReadOnlyEditor(EditorWithLabelComponent):
+    """An Editor which displays a label using the font."""
+
+    def init(self, parent):
+        self.font = self.value
+        super().init(parent)
+
+    def update_editor(self):
+        self.font = self.value
+        super().update_editor()
```

Comment:
Any reason not to use `self.value` in `EditorWithLabelComponent` rather than introducing a new trait?

Reply:
Yes, the idea is that in `EditorWithLabel` the value could be anything, and the font is independent of it (basically, it is styling in the base class).  In the font editor, we then add the constraint that the value and the font are the same.